Wrappers/Python/cil/optimisation/utilities/callbacks.py

```diff
@@ -187,3 +191,45 @@ def __call__(self, algorithm):
             raise StopIteration
             
         
+class SaveIterates(Callback):
+    r'''Callback to save iterates as tiff files every set number of iterations.  
+    
+    Parameters
+    ----------
+    interval: integer, 
+        The iterates will be saved every `interval` number of iterations e.g. if `interval =4` the 0, 4, 8, 12,... iterates will be saved. 
+    file_name : string
+        This defines the file name prefix, i.e. the file name without the extension.
+    dir_path : string
+        The place to store the images 
+    roi: dict, optional default is None and no slicing will be applied
+        The region-of-interest to slice {'axis_name1':(start,stop,step), 'axis_name2':(start,stop,step)}
+        The `key` being the axis name to apply the processor to, the `value` holding a tuple containing the ROI description
+        Start: Starting index of input data. Must be an integer, or `None` defaults to index 0.
+        Stop: Stopping index of input data. Must be an integer, or `None` defaults to index N.
+        Step: Number of pixels to average together. Must be an integer or `None` defaults to 1.
+    compression : str, default None. Accepted values None, 'uint8', 'uint16'
+        The lossy compression to apply. The default None will not compress data.
+        uint8' or 'unit16' will compress to unsigned int 8 and 16 bit respectively.
+    '''
+    def __init__(self, interval=1, file_name='iter',  dir_path='./', roi=None, compression=None): 
+
+        self.file_path= os.path.join(dir_path, file_name)
+            
+        self.interval=interval
+        self.roi=roi 
+        if self.roi is not None:
+            self.slicer= Slicer(roi=self.roi)
+        self.compression=compression
+        super(SaveIterates, self).__init__()  
+
+    def __call__(self, algo):
+        
+        if algo.iteration % self.interval ==0:
+            if self.roi is None:
+                TIFFWriter(data=algo.solution, file_name=self.file_path+f'_{algo.iteration:04d}.tif', counter_offset=-1,compression=self.compression ).write()
+            else:
+                self.slicer.set_input(algo.solution)
+                TIFFWriter(self.slicer.get_output(), file_name=self.file_path+f'_{algo.iteration:04d}.tif', counter_offset=-1,compression=self.compression ).write()
```

Comment:
btw is this meant to be `.tif` or `.tiff`?

Reply:
Pass - google says either is fine and I think the TIFFWriter strips it anyway:

https://github.com/MargaretDuff/CIL-margaret/blob/435e93fe702109c4875313c4d9abf7877963e374/Wrappers/Python/cil/io/TIFF.py#L106-L110

I will make it `tiff` to match the output of the TIFFWriter. Good catch